apis/workstations/v1alpha1/workstationconfig_types.go

```diff
@@ -0,0 +1,467 @@
+	//  this field is not set, the disks are encrypted using a generated
+	//  key. Customer-managed encryption keys do not protect disk metadata.
+	//
+	//  If the customer-managed encryption key is rotated, when the workstation
+	//  instance is stopped, the system attempts to recreate the
+	//  persistent disk with the new version of the key. Be sure to keep
+	//  older versions of the key until the persistent disk is recreated.
+	//  Otherwise, data on the persistent disk might be lost.
+	//
+	//  If the encryption key is revoked, the workstation session automatically
+	//  stops within 7 hours.
+	//
+	//  Immutable after the workstation configuration is created.
+	EncryptionKey *WorkstationConfig_CustomerEncryptionKey `json:"encryptionKey,omitempty"`
+
+	// Optional. Readiness checks to perform when starting a workstation using
+	//  this workstation configuration. Mark a workstation as running only after
+	//  all specified readiness checks return 200 status codes.
+	ReadinessChecks []WorkstationConfig_ReadinessCheck `json:"readinessChecks,omitempty"`
+
+	// Optional. Immutable. Specifies the zones used to replicate the VM and disk
+	//  resources within the region. If set, exactly two zones within the
+	//  workstation cluster's region must be specified—for example,
+	//  `['us-central1-a', 'us-central1-f']`. If this field is empty, two default
+	//  zones within the region are used.
+	//
+	//  Immutable after the workstation configuration is created.
+	ReplicaZones []string `json:"replicaZones,omitempty"`
+}
+
+// WorkstationConfigStatus defines the config connector machine state of WorkstationConfig
+type WorkstationConfigStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the WorkstationConfig resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *WorkstationConfigObservedState `json:"observedState,omitempty"`
+}
+
+// WorkstationConfigSpec defines the desired state of WorkstationConfig
+// +kcc:proto=google.cloud.workstations.v1.WorkstationConfig
+type WorkstationConfigObservedState struct {
+	// Output only. A system-assigned unique identifier for this workstation
+	//  configuration.
+	UID *string `json:"uid,omitempty"`
+
+	// Output only. Indicates whether this workstation configuration is currently
+	//  being updated to match its intended state.
+	Reconciling *bool `json:"reconciling,omitempty"`
```

Comment:
Nit: we probably have this in status.conditions

Reply:
Fixed!

Reply:
Accidentally just removed this from WorkstationCluster also, whoops. Will include that in this change as well.